design/jira-traceability.md

```diff
@@ -0,0 +1,150 @@
+
+
+
+## YAML: 
+
+Since traceability primarily involves interacting with external systems, we propose taking configuration for interaction with these external systems in a separate `integrations` section in YAML. As a beginning, we will support a first class integration with JIRA. In future, we may eventually make this section extensible thereby allowing 3rd party authors to create their own integrations. In this document, we will keep focus on JIRA integration. <em>Details on extensibility design are out of scope for this document.</em>
+
+ 
+
+### Schema:
+
+```yaml
+integrations:       
+- integration: jira # name of integration; mandatory;    
+  connection: string # service connection to JIRA cloud; mandatory;
+  resources: # resource filters. Status will be posted to JIRA for issues that are linked to these resources only; optional; defaults to all resources
+  - repository: string #  identifier of the repository resource as defined in YAML resources section
+    branches: 
+      include: [ string ] # optional; defaults to all branches.
+      exclude: [ string ] # optional; 
+  - pipeline: string #  identifier of the pipeline resource as defined in YAML resources section
+    branches:
+      include: [ string ]  # optional; defaults to all branches.
+      exclude: [ string ]  # optional;
+  - container: string #  identifier of the container resource as defined in YAML resources section
+    tags:
+      include: [ string ] # optional; defaults to all tags
+      exclude: [ string ] # optional;
+  environments: # environment filters. Status of deployments to only these environments will be  posted to JIRA; defaults to all environments
+  - environment: string #name of environment
+    type: string #environment type based on Atlassian's categorization of environment(Eg, production, staging etc). Defaults to unmapped.    
+```
+
+<br/>
+<br/>
+
+
+### Examples:
+
+Day-1 configuration: The following yaml snippet is the simplest configuration and eliminates all optional fields. The status will be posted back to JIRA server for all branches of all resources consumed in any environment
+
+```yaml
+integrations:       
+- integration: jira 
+  connection: myJiraServiceConnection 
+```
+
+<br/>
+<br/>
+
+The following configuration will send status to all JIRA WITs linked in the commits of `zenithWorks-kernel` repository resource from `releases/*` branch only. 
+
+```yaml
+integrations:       
+- integration: jira 
+  connection: myJiraServiceConnection 
+  resources: 
+  - repository: zenithWorks-kernel 
+    branches:
+      include: 
+      - releases/*   
+```
+
+<br/>
+<br/>
+
+The following configuration will post status to JIRA for images from `zenithworksImageRegistry` container registry that are tagged with `prod-ready`
+
+```yaml
+integrations:       
+- integration: jira 
+  connection: myJiraServiceConnection 
+  resources: 
+  - container: zenithworksImageRegistry
```

Comment:
is this means, if the deployment job has not used this container, we will not publish the data? same with pipeline or with repository?

Reply:
Yes. If the deployment job consumes this container, only then we publish data. Similar logic for other resource types as well.